tests/test_ranges.py

```diff
@@ -624,6 +624,7 @@ def test_does_not_modify_ranges(self):
     )
     def test_returns_true_if_and_ranges_overlap(self, ranges_):
         assert ranges.any_overlapping(ranges_)
+        assert ranges.any_overlapping(reversed(ranges_))
```

Comment:
This should probably have been part of https://github.com/octoenergy/xocto/pull/184/